### PR TITLE
 UI: add InfoBox component

### PR DESCRIPTION
**Pull request by alexanderzobnin**
This is open source part of the improving reporting UX when image renderer is not available/installed. We need `InfoBox` component (similar to one we use in data source configuration, but it's not a react component) to display message to user.

Screenshot from 2020-04-22 11-19-23

Screenshot from 2020-04-22 11-19-38

Also, this makes possible to use `ButtonHTMLAttributes` for `LinkButton`. In particular I need this to add `disabled` attribute to the LinkButton (it's possible for Button, but not for LinkButton - that's strange).

### CODE REVIEW

File: packages/grafana-ui/src/components/Logs/LogDetailsRow.tsx [hunk 1]
```diff
@@ -126,14 +126,13 @@ class UnThemedLogDetailsRow extends PureComponent<Props, State> {
                       href={link.href}
                       target={'_blank'}
                       onClick={
-                        link.onClick
-                          ? event => {
```
Comment (alexanderzobnin) [inline packages/grafana-ui/src/components/Logs/LogDetailsRow.tsx:130]: This caused TS checks failed with "implicit any" error.

**Review by marefr — APPROVED** [review]
I'm no frontend expert, but LGTM to me when the build succeeds

File: packages/grafana-ui/src/components/InfoBox/InfoBox.story.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { number } from '@storybook/addon-knobs';
+import { InfoBox } from './InfoBox';
+
+const InfoBoxStories = storiesOf('General/InfoBox', module);
```
Comment (torkelo) [inline packages/grafana-ui/src/components/InfoBox/InfoBox.story.tsx:6]: this is the old style of defining stories: please use https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/ThemeColors/ThemeColors.story.tsx#L4

File: packages/grafana-ui/src/components/InfoBox/InfoBox.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import { css, cx } from 'emotion';
+import { stylesFactory, useTheme } from '../../themes';
+
+export interface InfoBoxProps {
+  header?: string | JSX.Element;
+  footer?: string | JSX.Element;
+  className?: string;
+}
+
+const getInfoBoxStyles = stylesFactory((theme: GrafanaTheme) => ({
```
Comment (torkelo) [inline packages/grafana-ui/src/components/InfoBox/InfoBox.tsx:12]: Move styles to the bottom of file

File: packages/grafana-ui/src/components/InfoBox/InfoBox.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { GrafanaTheme } from '@grafana/data';
+import { css, cx } from 'emotion';
+import { stylesFactory, useTheme } from '../../themes';
+
+export interface InfoBoxProps {
```
Comment (torkelo) [inline packages/grafana-ui/src/components/InfoBox/InfoBox.tsx:6]: just name this type Props   (very rare cases where we have multiple components in a file do we prefix the props type)

File: packages/grafana-ui/src/components/Logs/LogDetailsRow.tsx (hunk 1 above)
Comment (dprokop) [reply packages/grafana-ui/src/components/Logs/LogDetailsRow.tsx:130]: you could check if onClick exists this way:
```
onClick={() => {
  if(onClick) {
    ...
```

File: packages/grafana-ui/src/components/InfoBox/InfoBox.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  `,
+  header: css`
+    margin-bottom: ${theme.spacing.d};
+  `,
+  footer: css`
+    margin-top: ${theme.spacing.d};
+  `,
+}));
+
+export const InfoBox: React.FunctionComponent<InfoBoxProps> = ({ header, footer, className, children }) => {
```
Comment (torkelo) [inline packages/grafana-ui/src/components/InfoBox/InfoBox.tsx:56]: /**
 * This is a simple InfoBox component, the api is not considered stable yet and will likely see breaking changes in future minor releases 
 * @alpha
 */

File: packages/grafana-ui/src/components/InfoBox/InfoBox.story.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      max: 1500,
+      step: 100,
+    },
+    CONTAINER_GROUP
+  );
+
+  return { containerWidth };
+};
+
+InfoBoxStories.add('default', () => {
```
Comment (torkelo) [inline packages/grafana-ui/src/components/InfoBox/InfoBox.story.tsx]: just export a story like this;: https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/ThemeColors/ThemeColors.story.tsx#L13

**Review by torkelo — APPROVED** [review]
Looks good but added some change suggestions

File: packages/grafana-ui/src/components/InfoBox/InfoBox.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { GrafanaTheme } from '@grafana/data';
+import { css, cx } from 'emotion';
+import { stylesFactory, useTheme } from '../../themes';
+
+export interface InfoBoxProps {
+  header?: string | JSX.Element;
+  footer?: string | JSX.Element;
+  className?: string;
```
Comment (torkelo) [inline packages/grafana-ui/src/components/InfoBox/InfoBox.tsx:9]: use React.forwardRef instead (with Div element),then no need to add className here

File: packages/grafana-ui/src/components/InfoBox/InfoBox.tsx (hunk 5 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/InfoBox/InfoBox.tsx:56]: use React.memo  to make this into pure component

Comment (torkelo) [reply packages/grafana-ui/src/components/InfoBox/InfoBox.tsx:56]: and use React.forwardRef

File: packages/grafana-ui/src/components/InfoBox/InfoBox.tsx (hunk 4 above)
Comment (alexanderzobnin) [reply packages/grafana-ui/src/components/InfoBox/InfoBox.tsx:6]: Yes, agreed

File: packages/grafana-ui/src/components/InfoBox/InfoBox.tsx (hunk 7 above)
Comment (alexanderzobnin) [reply packages/grafana-ui/src/components/InfoBox/InfoBox.tsx:9]: Hm, I'm not sure how `forwardRef` can help here if I need to pass `className` and merge it with existing styles? I'm looking into `Button` sources and there's className there:
https://github.com/grafana/grafana/blob/49a90cd764f1a57d6e03205747547d93525e27b0/packages/grafana-ui/src/components/Button/Button.tsx#L133-L145

Comment (torkelo) [reply packages/grafana-ui/src/components/InfoBox/InfoBox.tsx:9]: className is already defined on the element you are forward reffing: 

example here: https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/IconButton/IconButton.tsx#L11

Comment (alexanderzobnin) [reply packages/grafana-ui/src/components/InfoBox/InfoBox.tsx:9]: Ah, right, got it!

File: packages/grafana-ui/src/components/InfoBox/InfoBox.tsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  header?: string | JSX.Element;
+  footer?: string | JSX.Element;
+}
+
+/**
+ * This is a simple InfoBox component, the api is not considered stable yet and will likely see breaking changes
+ * in future minor releases.
+ * @Alpha
+ */
+export const InfoBox = React.memo(
```
Comment (alexanderzobnin) [inline packages/grafana-ui/src/components/InfoBox/InfoBox.tsx:16]: @torkelo could you confirm this is right? (memo wrapper before forwardRef)

Comment (torkelo) [reply packages/grafana-ui/src/components/InfoBox/InfoBox.tsx:16]: looks good

**Review by torkelo — APPROVED** [review]

**Pull request merged**